contrib/raftexample/kvstore.go

```diff
@@ -61,7 +65,18 @@ func (s *kvstore) readCommits(commitC <-chan *string, errorC <-chan error) {
 	for data := range commitC {
 		if data == nil {
 			// done replaying log; new data incoming
-			return
+			// OR signaled to load snapshot
+			snapshot, err := s.snapshotter.Load()
+			if err != nil && err != snap.ErrNoSnapshot {
+				log.Panic(err)
+			}
+			if err == snap.ErrNoSnapshot {
```

Comment:
probably move this before line 70? then we do not need to check err != errnosnapshot again when err is not nil